python/ray/function_manager.py

```diff
@@ -31,6 +33,221 @@
+            function_name = function_descriptor_list[2].decode()
+            if len(function_descriptor_list) == 4:
+                return cls(module_name, function_name, class_name,
+                           function_descriptor_list[3])
+            else:
+                return cls(module_name, function_name, class_name)
+        else:
+            raise Exception(
+                "Invalid input for FunctionDescriptor.from_bytes_list")
+
+    @classmethod
+    def from_function(cls, function, function_class=None):
+        """Create a FunctionDescriptor from a function instance.
+
+        This function is used to create the function descriptor from
+        a python function.
+
+        Args:
+            cls: Current class which is required argument for classmethod.
+            function: the python function used to create the function
+                descriptor.
+            function_class: The class that the function belongs to, which
+                could be None when this function is not a method of class.
+
+        Returns:
+            The FunctionDescriptor instance created according to the function.
+        """
+        module_name = function.__module__
+        function_name = function.__name__
+        if function_class is not None:
+            class_name = function_class.__name__
+        else:
+            class_name = ""
+
+        function_source_hasher = hashlib.sha1()
+        try:
+            # If we are running a script or are in IPython, include the source
+            # code in the hash.
+            source = inspect.getsource(function).encode("ascii")
+            function_source_hasher.update(source)
+            function_source_hash = function_source_hasher.digest()
+        except (IOError, OSError, TypeError):
+            # Source code may not be available:
+            # e.g. Cython or Python interpreter.
+            function_source_hash = b""
+
+        return cls(module_name, function_name, class_name,
+                   function_source_hash)
+
+    @classmethod
+    def from_class(cls, target_class):
+        """Create a FunctionDescriptor from a class.
+
+        Args:
+            cls: Current class which is required argument for classmethod.
+            target_class: the python class used to create the function
+                descriptor.
+
+        Returns:
+            The FunctionDescriptor instance created according to the class.
+        """
+        module_name = target_class.__module__
+        class_name = target_class.__name__
+        return cls(module_name, "", class_name)
+
+    @classmethod
+    def for_driver_task(cls):
+        """Create a FunctionDescriptor instance for a driver task."""
+        return cls("", "", "", b"")
+
+    @property
+    def is_for_driver_task(self):
+        """bool: whether this function descriptor is for driver task."""
+        return (len(self.module_name) == 0  and len(self.class_name) == 0
+                    and len(self.function_name) == 0)
```

Comment:
nit, a bit more pythonic:
`return all(len(x) == 0 for x in [self.module_name, self.class_name, self.function_name])`